code/addons/a11y/src/a11yRunner.ts

```diff
@@ -37,29 +40,58 @@ const runNext = async () => {
   runNext();
 };
 
-export const run = async (input: A11yParameters = defaultParameters) => {
+export const run = async (input: A11yParameters = DEFAULT_PARAMETERS) => {
   const { default: axe } = await import('axe-core');
 
-  const { element = 'body', config = {}, options = {} } = input;
-  const htmlElement = document.querySelector(element as string) ?? document.body;
+  const { config = {}, options = {} } = input;
 
-  if (!htmlElement) {
-    return;
+  // @ts-expect-error - the whole point of this is to error if 'element' is passed
+  if (input.element) {
+    throw new ElementA11yParameterError();
+  }
+
+  const context: ContextSpec = {
+    include: document?.body,
+    exclude: ['.sb-wrapper', '#storybook-docs'], // Internal Storybook elements that are always in the document
+  };
+
+  if (input.context) {
+    // 1. if context exists, but it's not an object with include or exclude, it's an implicit include to be used directly
+    if (
+      !(
+        typeof input.context === 'object' &&
+        ('include' in input.context || 'exclude' in input.context)
+      )
+    ) {
+      context.include = input.context as ContextProp;
+    } else if (typeof input.context === 'object' && 'include' in input.context) {
+      // 2. if context.include exists, use it
+      context.include = input.context.include as ContextProp;
+    }
```

Comment:
I can't help but feel these conditionals can be simplified. I generally don't like a negated if with an else case, as that implies it could be flipped to avoid the negation. Maybe that's not true here though? Maybe start by avoiding the repeated typeof check.

Reply:
yeah I agree this is not great, part of the problem is trying to satisfy TS. I've tried to make it slightly more readable now though.